src/HomogeneousCalorimeter_geo.cpp

```diff
@@ -288,38 +303,67 @@ static std::tuple<int, std::pair<int, int>> add_12surface_disk(Detector& desc, A
-  // row and column ID start from the top left corner
-  //
-  int row = 0, column = 0;
-  int N_row =  std::round((maxl_ptsy[0].y() - minl_ptsy[0].y()) / modSize.y());
-  int N_column =  std::round((maxl_ptsx[0].x() - minl_ptsx[0].x()) / modSize.x());
+  for (xml_coll_t position_i(pts_extrudedpolygon, _U(position)); position_i; ++position_i){
+    xml_comp_t position_comp = position_i;
+    epic::geo::Point inpt = {position_comp.x(), position_comp.y()};
+    In_vertices.push_back(inpt);
+  }
+
+  double minX = 0., maxX = 0., minY = 0., maxY = 0.;
+  for (auto &square : test_points) {
+    epic::geo::Point box[4] = {{square.x() + half_modx, square.y() + half_mody}, {square.x() - half_modx, square.y() + half_mody}, {square.x() - half_modx, square.y() - half_mody}, {square.x() + half_modx, square.y() - half_mody}};
+    if (epic::geo::isBoxInsidePolygon(box, Out_vertices)) {
+      if( square.x() < minX ) minX = square.x();
+      if( square.y() < minY ) minY = square.x();
+      if( square.x() > maxX ) maxX = square.x();
+      if( square.y() > maxY ) maxY = square.x();
+    }
+  }
 
+  int total_count = 0;
+  int row = 0, column = 0;
+  int N_row = std::round((maxY - minY) / modSize.y());
+  int N_column = std::round((maxX - minX) / modSize.x());
   auto rowcolumn = std::make_pair(N_row, N_column);
 
-  for (auto& p : points) {
-    column = std::round((p.x() - minl_ptsx[0].x()) / modSize.x());
-    row = std::round((maxl_ptsy[0].y() - p.y()) / modSize.y());
+  for (auto &square : test_points) {
+    epic::geo::Point box[4] = {{square.x() + half_modx, square.y() + half_mody}, {square.x() - half_modx, square.y() + half_mody}, {square.x() - half_modx, square.y() - half_mody}, {square.x() + half_modx, square.y() - half_mody}};
+
+    if (epic::geo::isBoxInsidePolygon(box, Out_vertices)) {
+      if(!epic::geo::isBoxInsidePolygon(box, In_vertices)) {
+        // std::cout << "Square inside the inner frame: " << square.x() << ", " << square.y() << std::endl;
+        column = std::round((square.x() - minX) / modSize.x());
+        row = std::round((maxY - square.y()) / modSize.y());
+        Transform3D tr_local = RotationZYX(Nrot, 0.0, 0.0) * Translation3D(square.x(), square.y(), 0.0);
+        auto modPV = (has_envelope ? env_vol.placeVolume(modVol, tr_local) : env.placeVolume(modVol, tr_global * tr_local));
+        modPV.addPhysVolID("sector", sector_id).addPhysVolID("row", row).addPhysVolID("column", column);
+        total_count++;
+      }
+    }
 
-    Transform3D tr_local = RotationZYX(Nrot, 0.0, 0.0) * Translation3D(p.x(), p.y(), 0.0);
-    auto modPV = (has_envelope ? env_vol.placeVolume(modVol, tr_local) : env.placeVolume(modVol, tr_global * tr_local));
-    modPV.addPhysVolID("sector", sector_id).addPhysVolID("row", row).addPhysVolID("column", column);
   }
 
+  std::cout << "New Min: " << minX << ", " << minY << " || Max: " << maxX << ", " << maxY << std::endl;
+  std::cout << "Total count new: " << total_count << std::endl;
+
```

Comment:
No output by default. Use `Printout` to `DEBUG`, e.g. https://github.com/eic/epic/blob/62f3a3803a9b95b9e367adc1f6391ef43ee98bfb/src/MPGDDIRC_geo.cpp#L210, then you can choose to run with log level set to debug.

Reply:
Got it. change to the printout in the last commit